File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        minpri = float('inf')
        maxpro = 0;
        for i in range(len(prices)):
            if prices[i] < minpri :
                minpri = prices[i];
            pro = prices[i]-minpri;
            if pro > maxpro:
                maxpro = pro;
        return maxpro;
        """
        :type prices: List[int]
        :rtype: int
        """
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minpri = Integer.MAX_VALUE;
        int maxpro = 0;
        for(int i=0;i<prices.length;i++){
            if(prices[i] < minpri){
                minpri = prices[i];
            }
            int pro = prices[i] - minpri;
            if(pro > maxpro){
                maxpro = pro;
            }
        }
        return maxpro;
        
        
    }
}